apis/oauth/src/main/java/org/jclouds/oauth/v2/config/OAuthProperties.java

```diff
@@ -36,6 +36,35 @@
     */
    public static final String CREDENTIAL_TYPE = "jclouds.oauth.credential-type";
 
+   /**
+    * When using oauth with Azure Active Direction and Client Credentials, a "resource" must
+    * be specified as part of the request.
+    *
+    * @see <a href="https://msdn.microsoft.com/en-us/library/azure/dn645543.aspx">doc</a>
+    */
+   public static final String RESOURCE = "jclouds.oauth.resource";
+
+   /**
+    * When using oauth with Azure Active Direction and Client Credentials, an "azure-credential"
+    * is a password.  Only used for OAuth Live Tests.
+    */
+   public static final String AZURE_CREDENTIAL = "oauth.azure-credential";
+
+   /**
+    * When using oauth with Azure Active Direction and Client Credentials, an "azure-identity"
+    * is the GUID associated with an application given permssion to perform operations in
+    * Azure Resource Manager.  Only used for OAuth Live Tests.
+    */
+   public static final String AZURE_IDENTITY = "oauth.azure-identity";
+
+   /**
+    * When using oauth with Azure Active Direction and Client Credentials, the azure-endpoint
+    * is the URL https://login.microsoftonline.com/<Tenant ID>/oauth2/token, where <Tenant ID>
+    * corresponds to your Azure AD Tenant.  Only used for OAuth Live Tests.
+    */
+   public static final String AZURE_ENDPOINT = "oauth.azure-endpoint";
```

Comment:
Perhaps I'm missing something here, but we just need to run the tests against one provider at a time. When running against Azure, users will set the azure credentials, endpoint *and the credential type property*. When testing against Google, they will set the Google credentials, and so when testing with DigitalOcean.

The key thing is to set the credential type, to let jclouds pick the right authentication flow. Setting it should allow to run the tests against the desired provider without issues... Unless I'm missing something? (It's been a while now since I played with the OAuth api internals)